test/mpi/f08/pt2pt/irsendf08.f90

```diff
@@ -78,15 +79,19 @@ subroutine test_pair_irsend( comm, errs )
          call MPI_Irsend(send_buf, count, MPI_REAL, next, tag, &
       &                   comm, requests(2), ierr)
 !
-         index = -1
-         do while (index .ne. 1)
-            call MPI_Waitany(2, requests, index, statuses, ierr)
+         completed = 0
+         do while (completed .lt. 2)
+            call MPI_Waitany(2, requests, index, statuses(1),ierr)
+            if (index .lt. 1 .or. index .gt. 2) then
+                print *, "Invalid index in MPI_Waitany:", index
+            end if
+            if (index .eq. 1) then
```

Comment:
The check needs to be inside the loop here to make sure to test the correct request.

Reply:
This change can be a separate commit. Could you combine all the other commits into a single commit as they are semantically the same change?

Reply:
Done